qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -391,3 +440,20 @@ def _compose_transforms(basis_transforms, source_basis, source_dag):
                 )
 
     return mapped_instrs
+
+
+def _compute_incomplete_basis(target):
+    incomplete_basis_gates = []
+    size_dict = defaultdict(lambda: 0)
+    size_dict[1] = target.num_qubits
+    for qarg in target.qargs:
+        if len(qarg) == 1:
+            continue
+        size_dict[len(qarg)] += 1
+    for inst, qargs in target.items():
+        qarg_sample = next(iter(qargs))
+        if qarg_sample is None:
+            continue
```

Comment:
What circumstance can `qarg_sample` be `None` rather than raising `StopIteration` here?

Reply:
There is no requirement that `qarg_sample` has to be `None` for all gates in target. I would assume in all cases it would, because that normally implies an ideal simulator (so there are no limits on connectivity or duration or error rates associated with the gate). But it's defined on a per instruction basis, so it's possible for someone to do something like:

```
target = Target(num_qubits=2)
target.add_instruction(XGate())  # qarg_sample is None
target.add_instruction(CXGate(), {(0, 1), (1, 0): None})  # qarg_sample is (0, 1)
```

Reply:
Ah, I didn't know about an empty qubit properties getting promoted to `{None: None}` - I'd assumed it would get promoted to `{}`.